### PR TITLE
machines: stop filtering out non 'dir' type pools from add disk dialog

### PR DESCRIPTION
**Pull request by KKoukiou**
The testAddDisk was refactored to be modular and two tests where added for testing adding disks on nfs and iscsi.
Ideally all pools types should be tested but since we are adding the disks with pool-volume in the XML attributes,  there is not difference on  how we handle different pool types.

### CODE REVIEW

**Comment by KKoukiou** [comment]
Needswork because creation of the disk on NFS sometimes times out, see https://fedorapeople.org/groups/cockpit/logs/pull-12120-20190620-140909-6d970f30-verify-debian-testing/log.html#70

**Comment by KKoukiou** [comment]
Locally all test passed, tried multiple distros as well.
Please don't review till the tests finish here as well.

**Comment by KKoukiou** [comment]
iscsi-pool-add-disk
nfs-pool-add-disk

File: test/verify/machineslib.py [hunk 1]
```diff
@@ -771,6 +794,16 @@ def verify_disk_added(self):
             expected_target='vdh',
         ).execute()
 
+        if "debian" not in m.image and "ubuntu" not in m.image:
+            # ISCSI driver does not support virStorageVolCreate API
```
Comment (martinpitt) [inline test/verify/machineslib.py]: This comment is confusing -- does that apply to libvirt in Debian/Ubuntu? or to some particular properties below?

Comment (KKoukiou) [reply test/verify/machineslib.py]: Oh, it was supposed to be explanation in why I used use_existing_volume='True' in the test bellow, and didn't test both scenarios (create new volume / use existing volume) as for example in nfs case.
libvirt virStorageVolCreateAPI is not capable to create volumes on iscsi pools, so such test wouldn't make sense.

Comment (KKoukiou) [reply test/verify/machineslib.py]: And of course it's not related to only debian/ubuntu, it's the same for all distros.

**Pull request merged**